tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AutoInstrumentationExtensions.cs

```diff
@@ -28,5 +32,25 @@ internal static void DisposeWithException(this Scope scope, Exception exception)
                 }
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void ServerlessDispose(this Scope scope)
+        {
+            if (scope != null)
+            {
+                scope.Dispose();
+                scope.Span.Context.TraceContext.CloseServerlessSpan();
+                try
+                {
+                    // here we need a sync flush, since the lambda environment can be destroy after each invocation
+                    // 3 seconds is enough to send payload to the extension (via localhost)
+                    Tracer.Instance.TracerManager.AgentWriter.FlushTracesAsync().Wait(TimeSpan.FromSeconds(ServerlessMaxWaitingFlushTime));
```

Comment:
Blocking on a async call seems suspicious and in some cases can cause deadlocks. @DataDog/apm-dotnet, any thoughts here?